### DISCUSSION TITLE
Write a Custom Recipe to modify constructors

### DISCUSSION THREAD

**Discussion by amit2103** [openrewrite/rewrite#3928] [General] [answered]
Hello we have a Spring Boot project which uses JDBC.

So we have certain constructors like 
```java
public Foo(NamedParameterJdbcOperations template) { initialization code }
```

we want to change it to
```java
public Foo(NamedParameterJdbcTemplate template ) {

add a extra instance variable for some custom stuff
}
```

So we want to change the type of the constructor argument and add an extra instance variable to do some stuff. 

Can anyone please guide me how to achieve this ?

**Comment by timtebeek** [comment, marked as answer]
Hi @amit2103 ; From your question it's not immediately clear to me what you have tried already, or where you are struggling. We have quite some docs on recipe authoring over at https://docs.openrewrite.org/authoring-recipes
In addition to that we have a [rewrite-recipe-starter](https://github.com/moderneinc/rewrite-recipe-starter) if you want to get started writing your own recipe.

Could you let me know which of the above resources you've already gone over, and where you have remaining questions?

**Reply by timtebeek** [reply]
Continued on [our Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA)
